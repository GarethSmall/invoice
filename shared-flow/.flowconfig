[ignore]
[include]

[libs]

[options]
module.system=haste
esproposal.decorators=ignore
esproposal.class_static_fields=enable
esproposal.class_instance_fields=enable

munge_underscores=true

suppress_type=$FlowIssue
suppress_type=$FlowFixMe
suppress_type=$FixMe

suppress_comment=\\(.\\|\n\\)*\\$FlowFixMe\\($\\|[^(]\\|(\\(>=0\\.\\(2[0-3]\\|1[0-9]\\|[0-9]\\).[0-9]\\)? *\\(site=[a-z,_]*react_native[a-z,_]*\\)?)\\)
suppress_comment=\\(.\\|\n\\)*\\$FlowIssue\\((\\(>=0\\.\\(2[0-3]\\|1[0-9]\\|[0-9]\\).[0-9]\\)? *\\(site=[a-z,_]*react_native[a-z,_]*\\)?)\\)?:? #[0-9]+
suppress_comment=\\(.\\|\n\\)*\\$FlowFixedInNextDeploy

module.file_ext=.js
module.file_ext=.json
module.file_ext=.jsx

[version]
0.67.1
