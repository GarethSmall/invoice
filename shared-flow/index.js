// @flow
export * from './src/Product';
export * from './src/Order';
export * from './src/OrderItem';