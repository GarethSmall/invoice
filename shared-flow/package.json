{
  "name": "shared-flow",
  "version": "1.0.0",
  "description": "Flow typing for our invoice",
  "main": "src/index.js",
  "scripts": {
    "flow": "flow check",
    "lint": "eslint --ext .js,.jsx ./src",
    "lint-fix": "eslint --ext js,jsx ./src --fix",
    "prettier": "prettier-eslint \"./src/**/*.{scss,js}\" --write"
  },
  "author": "Gareth Small",
  "license": "ISC",
  "devDependencies": {
    "babel": "^6.23.0",
    "babel-core": "^6.26.0",
    "babel-eslint": "^8.0.2",
    "babel-plugin-transform-class-properties": "^6.24.1",
    "babel-plugin-transform-decorators": "^6.24.1",
    "babel-plugin-transform-decorators-legacy": "^1.3.4",
    "babel-plugin-transform-es2015-modules-commonjs": "^6.26.0",
    "babel-plugin-transform-runtime": "^6.23.0",
    "babel-preset-es2015": "^6.24.1",
    "babel-preset-flow": "^6.23.0",
    "eslint": "^4.12.0",
    "eslint-config-prettier": "^2.9.0",
    "eslint-loader": "^1.9.0",
    "eslint-plugin-flow": "^2.29.1",
    "eslint-plugin-flowtype": "^2.46.1",
    "eslint-plugin-import": "^2.8.0",
    "eslint-plugin-jsx-a11y": "^6.0.2",
    "eslint-plugin-prettier": "^2.6.0",
    "flow-bin": "^0.67.1",
    "prettier-eslint-cli": "^4.7.1"
  }
}
