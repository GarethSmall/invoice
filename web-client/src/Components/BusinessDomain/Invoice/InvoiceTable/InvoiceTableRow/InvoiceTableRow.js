// @flow
import React, { PureComponent } from 'react';
import { OrderItem } from 'shared-flow';
import { moneyFormat } from '../../../../../Helpers/Math/math';

type Props = {
  item : OrderItem,
};
export default class InvoiceTableRow extends PureComponent<Props> {
  render() {
    const { item } = this.props;
    return (
      <tr>
        <td>{item.product.name}</td>
        <td>{item.quantity}</td>
        <td>{moneyFormat(item.product.price)}</td>
        <td>{moneyFormat(item.price)}</td>
        <td><button>Edit</button></td>
        <td><button>Delete</button></td>
      </tr>
    );
  }
}