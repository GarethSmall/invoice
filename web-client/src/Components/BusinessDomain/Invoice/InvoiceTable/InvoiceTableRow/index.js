//@flow
import InvoiceTableRow from './InvoiceTableRow';
export default InvoiceTableRow;
