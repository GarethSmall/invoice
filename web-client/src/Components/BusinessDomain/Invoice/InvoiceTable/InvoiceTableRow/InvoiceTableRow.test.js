// @flow
import React from 'react';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme from 'enzyme';
import shallow from 'enzyme/shallow';
import InvoiceTableRow from './';
import { OrderItem } from 'shared-flow';

const { describe, test, expect } = global;

Enzyme.configure({ adapter: new Adapter() });

describe('InvoiceTableRow', () => {
  test('should render', () => {
    const mockItem : OrderItem =  {
      _id: '1',
      order: '1',
      product: {
        _id: '1',
        name: 'Test T-shirt',
        price: 32.00,
      },
      quantity: 2,
      price: 64.00,
    };
    const component = shallow((
      <InvoiceTableRow item={mockItem} />
    ));
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });
});
