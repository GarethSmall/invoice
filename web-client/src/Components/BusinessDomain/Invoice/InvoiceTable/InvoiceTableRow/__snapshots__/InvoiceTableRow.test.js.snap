// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InvoiceTableRow should render 1`] = `
<tr>
  <td>
    Test T-shirt
  </td>
  <td>
    2
  </td>
  <td>
    $32.00
  </td>
  <td>
    $64.00
  </td>
  <td>
    <button>
      Edit
    </button>
  </td>
  <td>
    <button>
      Delete
    </button>
  </td>
</tr>
`;
